src/parquet/column/test-util.h

```diff
@@ -165,9 +166,108 @@ class DataPageBuilder {
   }
 };
 
+template <typename TYPE>
+class DictionaryPageBuilder {
+ public:
+  typedef typename TYPE::c_type TC;
+  static constexpr int TN = TYPE::type_num;
+
+  // This class writes data and metadata to the passed inputs
+  explicit DictionaryPageBuilder(InMemoryOutputStream* sink) :
+      sink_(sink),
+      num_dict_values_(0),
+      encoding_(Encoding::RLE_DICTIONARY),
+      have_values_(false) {
+  }
+
+  void AppendValues(const ColumnDescriptor *d, const vector<TC>& values,
+      Encoding::type encoding = Encoding::RLE_DICTIONARY) {
+    int type_length = 0;
+    dict_buffer_ = std::make_shared<OwnedMutableBuffer>();
+    rle_indices_ = std::make_shared<OwnedMutableBuffer>();
+    if (TN == Type::FIXED_LEN_BYTE_ARRAY) {
+      type_length = d->type_length();
+    }
+    DictEncoder<TC> encoder(&pool_, type_length);
+    ASSERT_NO_THROW(
+        {
+          for (int i = 0; i < values.size(); ++i) {
+            encoder.Put(values[i]);
+          }
+        });
+  // Dictionary encoding
+    dict_buffer_->Resize(encoder.dict_encoded_size());
+    encoder.WriteDict(dict_buffer_->mutable_data());
```

Comment:
Perhaps this can be refactored to return a buffer of RLE-encoded dict indices for each input chunk of values, then the dict can be written at the end for serialization purposes. This way it will be easier to add tests with multiple dict-encoded data pages (there may be new previously unobserved dict values from page to page)